compiler/rustc_const_eval/src/transform/check_consts/check.rs

```diff
@@ -690,6 +690,7 @@ impl<'tcx> Visitor<'tcx> for Checker<'_, 'tcx> {
             | StatementKind::StorageLive(_)
             | StatementKind::StorageDead(_)
             | StatementKind::Retag { .. }
+            | StatementKind::PlaceMention(..)
```

Comment:
You are right. This would introduce an extra diagnostic for "dereferencing raw mutable pointers in constant functions is unstable". Exactly like unsafeck.

```rust
const unsafe fn foo(x: *mut usize) {
    let _ = *x;
}
```

As everything else around raw mutable pointers is already forbidden, I don't think this can break code. Can it?
Added a test.